### ISSUE TITLE
Feature Request:  Support for SurePet Feeder Connect

### ISSUE THREAD

**Issue by LeviJSmith** [wylanswets/homebridge-sure-petcare-platform#2]
This is an awesome plugin.  Thanks so much for creating it.  If you could add support for the SurePet Feeder Connect, that would be great.

**Comment by wylanswets** [comment]
Hey @LeviJSmith. Thanks! I'm glad you like it.

How would you anticipate that feeder being implemented in homekit? I'm unfamiliar with how exactly it works (like what controls you might have). Can you lock it? Or are you mostly interested in knowing when a pet is eating (like an occupancy sensor)?

The other thing is since I do not own one of these I would need an invite to your household through the app so I can test it out and get any code working.

Let me know what you think!

**Comment by LeviJSmith** [comment]
Good point.  I really hadn’t thought of what functionality I would want.  An occupancy sensor that detects when the lid is open and they are eating would be cool.  Would also be cool to have it detect battery levels.  I’m cool inviting you thru the app to develop as well.  Let me know when and if you would like to proceed.

**Comment by wylanswets** [comment]
I could give that a go, sure! Just invite this email to your home so I can mess with it: wylan.swets+petcare@gmail.com

**Comment by LeviJSmith** [comment]
I added you and gave you view mode. Let me know if you need higher access.

**Comment by LeviJSmith** [comment]
Adding the battery indicator to the existing cat door functionality would be cool as well.

**Comment by wylanswets** [comment]
Thanks @LeviJSmith. So I am looking at the data coming back through the API. Through the web interface I can't "pull up" the feeders to see if the lid is open - however through the API I am seeing the door lock status for the pet flaps. The wording is strange on their API so it could be that they just have a strange word for "lid status". Basically it would be cool if there's some way we could coordinate having the door open so I can pull the API and compare the difference? Maybe we can connect on Discord or something.

That said too - it may be helpful if I can pull up the device in the web control which may require a higher level of access unfortunately.

Lastly on battery - I'm still trying to figure out how they determine battery level. I thought I understood but their number representations are strange... 6.108 and 6.121 is 100% but 5.334 and 5.182 are both 50% lol.... so it would be helpful if I can pull the API when one of your batteries goes to 25% and 75% so I can have a full picture to accurately report the battery percentage (even if it's just in those 4 steps).

**Comment by LeviJSmith** [comment]
Greetings,

I increased your access.  Let me know if that helps.  As for the battery question, Garage Patio Door is at full battery, Betty Feeder is at full battery, and Rigby feeder is at half battery.  Hope that helps.

-Levi

**Comment by wylanswets** [comment]
Excellent. Can you confirm for me that you see some kind of a status in the app when the door is open on the feeder?

**Comment by LeviJSmith** [comment]
Greetings,

I looked around and it doesnt look like there is any indicator in the app when the lid is open or closed.  Guessing that means a no go on that front?

-Levi

**Comment by wylanswets** [comment]
Bummer.... it’s possible that is the case... when you opened it was that with the pet chip / collar or just manually? Oh... and also I wonder if there’s a “feeding” status on the pet itself in the app? If it shows on the pet I would be able to pull it from there.

**Comment by LeviJSmith** [comment]
No status either manually pressing the open button or having the pet use the chip.  There is feeding activity screen on each pet that keeps track of when they eat and how much.  Will that work?

**Comment by wylanswets** [comment]
There would have to be some kind of indication that they are currently eating for me to put a sensor that shows they are currently eating.

**Comment by LeviJSmith** [comment]
Greetings.  I was thinking that might be the case.  No worries.  Maybe Sure will add something to the API at a later date.

Will you be able to do something with battery level?  That would be cool to have on the feeders as well as the pet door.

-Levi

**Comment by wylanswets** [comment]
Right, if they do I would be happy to add that in!

For the batteries, yes. As soon as one of your batteries drains to 25% I want to pull the API again to see if the pattern continues, and I believe the battery status is the same on all of their products. 2 things from there:

1) Does the status go lower than 25% or does it basically warn you of low battery at 25%? (Sorry I don't actually own any of these products myself). How long does it usually take to go from 50% to 25% (or drained to a warning status for that matter). Lastly... my thought is to just use 25% as the "low battery" indicator for homekit but let me know what your thoughts are.

2) I would have to at least make an "occupancy sensor" out of the feeders that would not report the status just so I could put the battery indicator on them. That would probably be a good idea anyway so if they add feeding status in the future we would already have the spot to pull the data from (or if one of my other ideas pans out which we can talk about later).

**Comment by LeviJSmith** [comment]
Greetings,

The battery indicator is a picture of the battery with 4 bars in it.  I agree that using 25% (one bar) as the low battery indicator makes sense.  I’ll let you know when one of my feeders reaches it.  One of them is at 50% now so I suspect it will change to 25% in the next few weeks. 

I’m cool with using the occupansy sensor as a placeholder to display battery.

-Levi

**Comment by wylanswets** [comment]
A FEW WEEKS? Dang. That's a good battery haha. Great, yeah as soon as it happens let me know! I'll check up on it here or there but I suspect you'll notice it first.

**Comment by LeviJSmith** [comment]
Greetings. It’s at 1 bar now if you want to take a look.

**Comment by LeviJSmith** [comment]
Well, it’s back to 2 bars. That’s weird, I’m guessing it’s just on the cusp.

**Comment by LeviJSmith** [comment]
Back down to 1.  I think we’re at the point we would want the low battery status here.

**Comment by LeviJSmith** [comment]
Greetings,

Did you get what you needed?  Am I ok to change the batteries?

-Levi

**Comment by wylanswets** [comment]
Hello @LeviJSmith,

I did - sorry I have not responded! Yes, go ahead and change the batteries. And by the way, what kind of batteries are they? They are disposable and replaceable batteries?

Thanks,
Wylan Swets

**Comment by LeviJSmith** [comment]
Greetings,

Glad to hear from you.  I was starting to get worried that you might have gotten sick.  Glad you are well.  I’ll change the batteries this afternoon.  It takes 4 C batteries.

-Levi

**Comment by LeviJSmith** [comment]
Batteries have been changed.

**Comment by wylanswets** [comment]
@LeviJSmith Sorry for the long delay again - so the good news is I just pushed up an update that finally supports the battery state of the pet doors! I do not yet have the battery state for the feeders (as I have yet to add those).

I will soon begin working on that so you can get a battery status of the feeders, and I was otherwise trying to think through some other fun ways that maybe we could get the amount of food in there but if not I'll just add in having a place to see the battery info in homekit.

**Comment by LeviJSmith** [comment]
Just ran the update.  Awesome to see battery status in the door. Appreciate all your doing to add the functionality for battery on the feeders.

**Comment by LeviJSmith** [comment]
Greetings.  Just checking in.  Have you had time to work on the other enhancements?

-Levi

**Comment by geoffgbsn** [comment]
I have 2 of these feeders, as well. I also have a Felaqua waterer. I'd love to see support for both but, as you've been discussing, there aren't a lot of options. Battery functionality would be nice. Besides consumption info, Felaqua sends notifications when it is running low as well as "refresh" reminders so that the waterer doesn't get nasty. Like the feeders, it takes 4 C batteries. If I can help by providing you more information from my 2 feeders and feklaqua by giving you access to my app, I'd be happy to. Would love to see something. Anything. The SurePet app is lacking in a lot.

**Comment by geoffgbsn** [comment]
As a matter of fact, I saw your email address up there and just went ahead and sent you and invite.

**Comment by booshtukka** [comment]
Would also like the Felaqua and Pet Feeder support. Mainly for battery levels, and for notifications i.e. water/food was eaten, water/food running low. :)
